ext/node/polyfills/punycode.ts

```diff
@@ -7,22 +7,33 @@ import {
   op_node_idna_punycode_encode,
 } from "ext:core/ops";
 
+import { deprecate } from "node:util";
+
 import { ucs2 } from "ext:deno_node/internal/idna.ts";
 
+function punyDeprecated(fn: any) {
+  return deprecate(
+    fn,
+    "The `punycode` module is deprecated. Please use a userland " +
+      "alternative instead.",
+    "DEP0040",
+  );
+}
+
 function toASCII(domain) {
-  return op_node_idna_domain_to_ascii(domain);
+  return punyDeprecated(op_node_idna_domain_to_ascii)(domain);
```

Comment:
Wrapping each function feels kinda gross, open to suggestions

Reply:
That seems reasonable 👍